lib/ts/recipe/session/recipeImplementation.ts

```diff
@@ -227,6 +246,91 @@ export default function getRecipeInterface(querier: Querier, config: TypeNormali
             }
         },
 
+        assertClaims: async function (
+            this: RecipeInterface,
+            input: {
+                session: SessionContainerInterface;
+
+                claimValidators: SessionClaimValidator[];
+                userContext?: any;
+            }
+        ): Promise<void> {
+            logDebugMessage("getSession: required validator ids " + input.claimValidators.map((c) => c.id).join(", "));
+            await input.session.assertClaims(input.claimValidators, input.userContext);
+            logDebugMessage("getSession: claim assertion successful");
+        },
```

Comment:
is this function being called from anywhere?

Reply:
Yeah: in `verifySession` and `getSession`.
It's trivial to replace with calls on `SessionContainer` but this is easier to override.